File: erp_mmust/erp_mmust/report/fee_collection_per_vote_summary/fee_collection_per_vote_summary.py
import frappe
from frappe import _

def execute(filters=None):
    if not filters:
        filters = {}

    columns = get_columns()
    data = get_data(filters)
    
    return columns, data, None, None, None

def get_columns():
    return [
        {
            "label": _("Student ID"),
            "fieldname": "student_id",
            "fieldtype": "Link",
            "options": "Customer",
            "width": 200
        },
        {
            "label": _("Student Name"),
            "fieldname": "student_name",
            "fieldtype": "Data",
            "width": 300
        },
        {
            "label": _("Amount"),
            "fieldname": "amount",
            "fieldtype": "Currency",
            "width": 150
        }
    ]

def get_data(filters):
    if not filters.get("account"):
        return []

    data = frappe.db.sql("""
        SELECT
            gle.party as student_id,
            c.customer_name as student_name,
            SUM(gle.credit) - SUM(gle.debit) as amount
        FROM `tabGL Entry` gle
        LEFT JOIN `tabCustomer` c ON gle.party = c.name
        WHERE
            gle.company = %(company)s
            AND gle.account = %(account)s
            AND gle.posting_date BETWEEN %(from_date)s AND %(to_date)s
            AND gle.party_type = 'Customer'
            AND c.customer_group = 'Student'
        GROUP BY
            gle.party, c.customer_name
        HAVING
            amount != 0
    """, values=filters, as_dict=True)

    return data

File: erp_mmust/erp_mmust/report/fee_collection_per_vote_summary/fee_collection_per_vote_summary.js
frappe.query_reports["Fee Collection Per Vote Summary"] = {
    "filters": [
        {
            "fieldname": "company",
            "label": __("Company"),
            "fieldtype": "Link",
            "options": "Company",
            "default": frappe.defaults.get_user_default("Company"),
            "reqd": 1
        },
        {
            "fieldname": "from_date",
            "label": __("From Date"),
            "fieldtype": "Date",
            "default": frappe.datetime.add_months(frappe.datetime.get_today(), -1),
            "reqd": 1
        },
        {
            "fieldname": "to_date",
            "label": __("To Date"),
            "fieldtype": "Date",
            "default": frappe.datetime.get_today(),
            "reqd": 1
        },
        {
            "fieldname": "parent_account",
            "label": __("Parent Vote (Account)"),
            "fieldtype": "Link",
            "options": "Account",
            "default": "41-00-000 - Revenue from Exchange Transactions - MMUST",
            "get_query": function () {
                return {
                    "filters": {
                        "is_group": 1,
                        "company": frappe.query_report.get_filter_value('company')
                    }
                };
            }
        }
    ]
};
